src/resolve/sourceComponent.ts

```diff
@@ -116,17 +116,45 @@ export class SourceComponent implements MetadataComponent {
   public async parseXml<T = JsonMap>(xmlFilePath?: string): Promise<T> {
     const xml = xmlFilePath ?? this.xml;
     if (xml) {
-      const contents = await this.tree.readFile(xml);
-      return this.parse<T>(contents.toString());
+      const contents = (await this.tree.readFile(xml)).toString();
+      try {
+        return this.parse<T>(contents);
+      } catch (e) {
+        // only attempt validating once there's an error to avoid the performance hit of validating every file
+        const validation = validate(contents);
+        if (validation !== true) {
+          throw new LibraryError('invalid_xml_parsing', [
+            xml,
+            validation.err.msg,
+            validation.err.line.toString(),
+            validation.err.code,
+          ]);
+        }
+        throw e;
+      }
     }
     return {} as T;
   }
 
   public parseXmlSync<T = JsonMap>(xmlFilePath?: string): T {
     const xml = xmlFilePath ?? this.xml;
     if (xml) {
-      const contents = this.tree.readFileSync(xml);
-      return this.parse<T>(contents.toString());
+      const contents = this.tree.readFileSync(xml).toString();
+      try {
+        return this.parse<T>(contents);
+      } catch (e) {
+        // only attempt validating once there's an error to avoid the performance hit of validating every file
+        const validation = validate(contents);
```

Comment:
I think having them both use a shared`private parseAndValidateXML()` would be more DRY.

Reply:
I had that for a while - but it was causing testing issues because `parse` was stubbed, but I got it working 👍